src/api/fetchSearch.ts

```diff
@@ -4,10 +4,16 @@ import { v4 as uuidv4 } from 'uuid';
 
 export async function fetchSearch({
```

Comment:
Please use return z[.parse](https://zod.dev/?id=parse) in this function, and then remove : Promise<ResponseAPI> from the function because TS will get types implicitly from the zod schema